tests/core/matrix.rs

```diff
@@ -1263,6 +1263,20 @@ fn column_iterator_double_ended_mut() {
     assert_eq!(col_iter_mut.next(), None);
 }
 
+#[test]
+fn test_inversion_failure_leaves_matrix_unchanged() {
+    let mut mat = na::Matrix4::new(
+        1.0, 2.0, 3.0, 4.0,
+        2.0, 4.0, 6.0, 8.0,
+        3.0, 6.0, 9.0, 12.0,
+        4.0, 8.0, 12.0, 16.0
+    );
+    let expected = mat.clone();
+    if !mat.try_inverse_mut() {
```

Comment:
This test is currently very brittle. Imagine that some change in an implementation detail somewhere caused `mat.determinant()` to return something not exactly `0.0`. Then `mat.try_inverse_mut()` might return `true` and the test would succeed, although it's no longer testing what we want to test it.

Perhaps replace the if by an assert? E.g. `assert!(!mat.try_inverse_mut())` followed by the `assert_eq` that you already have.

Reply:
@Andlon fixed it